### PR TITLE
Adds Ketamine

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request

Adds Ketamine as a Chemical to the Game. 
It is also included under the Junkie Trait.
Basically an Amped up version of microdosing morphine. Provided a mood boost, ignores damage slowdowns, will put your ass to sleep.
As a dissociative anasthetic(annasthetic? anascthetic?) **make you go unconscious drug**, there are also hallucinations involved. Overdosing is bad, very bad. You get a split personality from dissociating so hard, you hallucinate harder, you lose the speed buff. you become dizzier, you may just fall over and drop shit.

I had to do a lot of research to get a desireable list of effects.

### Making Ketamine:
3 units Morphine, 3 units Chloral Hydrate, 3 units Fentanyl, 3 units Epinephrine

## Why It's Good For The Game
Adding new recreational chem to the Junkie trait. Provides a new mix of effects from a single chem. New chem for chemists to produce and offer should they decide to go the drug dealing route.
Overall, variety is the spice of life.

## Changelog
:cl:
add: Ketamine, as a Narcotic
add: Ketamine to the list of Junkie reagents
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
			M.drowsyness ++
```

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:506]: This is not mood boost, if you are going for mood boost, create mood event and apply it here instead.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
	//Dissociative anesthetics? Overdosing? Time to dissociate hard.
```

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:515]: what

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: If he doesn't have that trait you are removing it?

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (Archanial) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:527]:
```suggestion
			to_chat(M, "<span class='warning'>You your limbs won't move...</span>")
```

**Review by Archanial — COMMENTED** [review]
This will also require wiki changes

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (DatBoiTim) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:506]: > This is not mood boost, if you are going for mood boost, create mood event and apply it here instead.

It's an indirect. Mostly addressing disgust. Meant to be weaker than happiness.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (DatBoiTim) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:515]: Ensures does not have split personality twice. Especially if multiple overdoses. Also ensures that the trauma is compatible.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (DatBoiTim) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Because I am so galaxy brained I fucked up the logic and lost the ability to spell and grammar.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (DatBoiTim) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:527]: Replaced grammar error with another.

**Comment by OceanFish1** [comment]
analsthetic

**Comment by utilbird** [comment]
I thought krokodil was supposed to be the fictional equivalent

**Comment by DatBoiTim** [comment]
> I thought krokodil was supposed to be the fictional equivalent 

Krokodil is a real thing and it's known for fucking your skin up pretty badly amongst its opiate style effects.

**Comment by DatBoiTim** [comment]
> This will also require wiki changes

Nothing I can't handle.

**Comment by TeomanTheGreat** [comment]
Merge when

**Comment by DatBoiTim** [comment]
Some point in the future?

**Comment by utilbird** [comment]
We'll never get that war robots sponsorship now...

**Comment by qwertyquerty** [comment]
Will merge after wiki changes are made

**Review by qwertyquerty — DISMISSED** [review]

**Comment by DatBoiTim** [comment]
> Will merge after wiki changes are made

https://wiki.beestation13.com/view/Guide_to_chemistry#Ketamine I can't figure out the background or the Reagent Color thing. But the important information is there.

File: code/modules/reagents/chemistry/recipes/drugs.dm
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/recipes/drugs.dm:55]: Why have it be 3 parts of everything instead of 1 part

Comment (DatBoiTim) [reply code/modules/reagents/chemistry/recipes/drugs.dm:55]: To encourage batch production. Also meant to require a larger amount for production.

Comment (utilbird) [reply code/modules/reagents/chemistry/recipes/drugs.dm:55]: I believe the recipe to result ratio is the exact same (4:1). If you wanted to increase the amount of required reagents to give a single unit of ketamine, just drop the result amount a bit :)

Comment (DatBoiTim) [reply code/modules/reagents/chemistry/recipes/drugs.dm:55]: Unless I misunderstood that means instead of a per unit(such as with oil), it takes 12 and collapses into 3

Comment (utilbird) [reply code/modules/reagents/chemistry/recipes/drugs.dm:55]: I'm not _entirely_ sure what you mean but I'll explain it :D 

in this case you've got a 12:3 recipe:result ratio, I was just pointing out how this is the same as a 4:1 ratio.

3 morphine  + 3 chloral + 3 fenatyl + 3 epi = 3 ketamine
is equal to
1 morphine + 1 chloral + 1 fenatyl + 1 epi = 1 ketamine

Soo there's no real reason to have it 3 for everything _unless_ you wanted to make the initial reagent amount require 12 instead of 4. If that's the case then your current code works fine, although I personally don't see why you'd want it to do that xD

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (park66665) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:515]:
```suggestion
	if(B.can_gain_trauma(/datum/brain_trauma/severe/split_personality, 5))
		B.brain_gain_trauma(/datum/brain_trauma/severe/split_personality, 5)
		. = 1
```

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (park66665) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
	..()
```

Looks a bit weird? But this is how `process_overdose` is meant to be written.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (park66665) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
	..()
```

Comment (DatBoiTim) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Yeah... if you took any amount of time looking at change history, it's very clear I was a mess

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (park66665) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]:
```suggestion
	M.hallucination = max(M.hallucination, 5)
```
`M.hallucination = 5` doesn't *add* hallucination, it will set hallucination to 5 no matter what. Hallucination of 5 isn't that strong, and it'll *override* other hallucinogens. You also can do this:

```suggestion
	M.hallucination += 5
```

**Review by park66665 — CHANGES_REQUESTED** [review]
This should be the last review before merge.

**Review by park66665 — APPROVED** [review]
I still am not sure about how difficult the synthesis should be. I don't know about how Ketamine works. (Sleeping part is great, though.) But I know that this PR is basically unstoppable atm. This PR is good to go.

**Pull request merged**
